instrumentation/opentelemetry-instrumentation-flask/tests/test_programmatic.py

```diff
@@ -250,6 +268,34 @@ def test_exclude_lists_from_explicit(self):
         span_list = self.memory_exporter.get_finished_spans()
         self.assertEqual(len(span_list), 1)
 
+    def test_flask_metrics(self):
+        self.client.get("/hello/123")
+        self.client.get("/hello/321")
+        self.client.get("/hello/756")
+        metrics_list = self.memory_metrics_reader.get_metrics_data()
+        number_data_point_seen = False
+        histogram_data_point_seen = False
+        self.assertTrue(len(metrics_list.resource_metrics) != 0)
+        for resource_metric in metrics_list.resource_metrics:
+            self.assertTrue(len(resource_metric.scope_metrics) != 0)
+            for scope_metric in resource_metric.scope_metrics:
+                self.assertTrue(len(scope_metric.metrics) != 0)
+                for metric in scope_metric.metrics:
+                    self.assertIn(metric.name, _expected_metric_names)
+                    data_points = list(metric.data.data_points)
+                    self.assertEqual(len(data_points), 1)
+                    for point in data_points:
+                        if isinstance(point, HistogramDataPoint):
+                            self.assertEqual(point.count, 3)
```

Comment:
Can we have a test for the actual values in the histogram? This will test that what we are recording for duration value is actually accurate.

Reply:
I'm sorry, I do not understand how can we get the value of time duration to test. Can you please give me an example?

Reply:
`point.sum` gives you the total duration.

Reply:
Yes, we can get the actual value, but what will be the expected value? For three get requests `point.sum` value will be random time taken. We can not compare it to some hardcoded value. One possible solution I can think of is to mock the `default_timer()` function that we use to get the timestamp, and with that, we test the time duration captured. But I'm not sure if it is the correct way to test?

Reply:
You do not have to make it exact hardcoded value. It can be `assertGreaterEqual` or `assertAlmostEqual` with delta based on how much time a view/endpoint takes to respond back.

Reply:
trigger a request -> operation takes some time -> expected duration should be within x+/-∆

Reply:
I've added the test. Please have a look.